Sources/Core/Utilities/Logger/Logger.swift

```diff
@@ -64,25 +64,47 @@ public extension Logger {
   static func error(_ message: String, params: Parameters? = nil, file: String = #file, function: String = #function, line: Int = #line) {
     flush(.error, message: message, params: params, file: file, function: function, line: line)
   }
-  
-  /// Log crash
-  static func crash(params: Parameters, file: String = #file, function: String = #function, line: Int = #line) {
-    flush(.error, message: "app_crash", params: params, file: file, function: function, line: line)
-  }
 }
 
 // MARK: - Private Methods
 private extension Logger {
   static func flush(_ level: LogLevel, message: String, params: Parameters? = nil, file: String, function: String, line: Int) {
     guard shared.logLevel.rawValue >= level.rawValue else { return }
     
-    let fileName = URL(fileURLWithPath: file).lastPathComponent.components(separatedBy: ".").first ?? ""
-    var messagePrint = "\(level.string): \(fileName).\(function)[\(line)]: \(message)"
-    if let params = params, !params.isEmpty {
-      messagePrint += ", params: \(params)"
+    let fileName = URL(string: file)?.lastPathComponent ?? ""
+    let nl = "\n ⮑ "
+    var messagePrint = "\(level.label): \(message)"
+    if line >= 0 {
+      messagePrint += "\(nl)source: \(fileName).\(function)[\(line)]"
+    }
+    if let params, !params.isEmpty {
+      let groupedParams = params
+        .map { "\($0.key): \($0.value)" }
+        .sorted()
+        .joined(separator: nl)
+      messagePrint += "\(nl)\(groupedParams)"
     }
     
-    debugPrint(messagePrint)
-    debugPrint()
+    if #available(iOS 14.0, *) {
+      let category = "\(fileName) - \(function) - line \(line)"
+      let logger = os.Logger(subsystem: Bundle.main.bundleIdentifier ?? "povioKit.logger", category: category)
+      switch level {
+      case .none:
+        break
+      case .error:
+        logger.error("\(messagePrint)")
```

Comment:
we can pass directly pass `messagePrint` without string interpolation

Reply:
Not really - https://developer.apple.com/documentation/os/oslogmessage